include/sys/dataset_kstats.h

```diff
@@ -60,10 +61,16 @@ typedef struct dataset_kstat_values {
 
 typedef struct dataset_kstats {
 	dataset_sum_stats_t dk_sums;
+	zil_sums_t dk_zil_sums;
 	kstat_t *dk_kstats;
 } dataset_kstats_t;
 
-void dataset_kstats_create(dataset_kstats_t *, objset_t *);
+typedef struct dataset_all_kstat_values {
+	dataset_kstat_values_t dkv;
+	zil_stats_t zkv;
+} dataset_all_kstat_values_t;
```

Comment:
This extra type here causes a bunch of additional churn that we could avoid if we just embed `zil_stats_t` into `dataset_kstat_value_t`. I don't see which adding this type makes things any cleaner, architecturally.